pkg/util/log/log.go

```diff
@@ -534,6 +534,15 @@ func Tracef(format string, params ...interface{}) {
 	logFormat(seelog.TraceLvl, func() { Tracef(format, params...) }, Logger.tracef, format, params...)
 }
 
+// TracefStackDepth logs with format at the trace level and the current stack depth plus the given depth
+func TracefStackDepth(format string, depth int, params ...interface{}) {
+	currentLevel, _ := GetLogLevel()
+	if currentLevel <= seelog.TraceLvl {
+		m := fmt.Sprintf(format, params...)
+		logContext(seelog.TraceLvl, func() { Tracec(m, nil) }, Logger.trace, m, depth, nil)
```

Comment:
Instead of using the context variant of each function (`Tracec`, `Debugc`, ...) with a nil context, you should probably use the base variant with stack depth (`TraceStackDepth`, `DebugStackDepth`, ...), and have something very similar to how it's implemented in the log component https://github.com/DataDog/datadog-agent/blob/443ac501027aecbc0cc754efc8055daaef2eedc3/comp/core/log/logger.go#L83-L88

```suggestion
		TraceStackDepth(1 + depth, fmt.Sprintf(format, params...))
```
(not super sure about `1 + depth`)